### DISCUSSION TITLE
Url Params in Data Source Settings

### DISCUSSION THREAD

**Discussion by mburamorde** [grafana/grafana-infinity-datasource#219] [Q&A]
Hi,

(very nice plugin, thanks for the good work!)

I'am getting crazy because of the following thing:
I thought I could add URL Params in the DataSource Settings, thereby the key is added always to every request.
![2021-12-23-15-57-02](https://user-images.githubusercontent.com/5478241/147257449-8c45207f-717e-4796-b782-863efc8f8d19.png)

But currently the "URL Param" api_key appears in no request.  what did I not understand.

Thanks,
Michael

**Comment by mburamorde** [comment]
My challenge is that every request should contain the api_key and I'am not want to enter it with every request.

**Reply by yesoreyeram** [reply]
I think you got it correctly. The settings can have any number of keys. You need to explicitly specify them in the query editor to consume them. That is by design. 

May be in future versions, we will include them by default.

**Comment by mburamorde** [comment]
Hi, thanks for the fast response! But how can I add the API Key via variable? Can I access the params from the data source settings? 

![20211224-090949](https://user-images.githubusercontent.com/5478241/147333118-edca5db4-3489-4f92-ac11-f15e3523c7e6.png)

**Reply by yesoreyeram** [reply]
Have you checked the docs here https://sriramajeyam.com/grafana-infinity-datasource/wiki/url ![Screenshot_20211224-105808](https://user-images.githubusercontent.com/153843/147347183-caa0bfa0-57fb-4f64-8d36-921a33d33378.png)

**Comment by mburamorde** [comment]
Thanks, very nice. It works  obviously ;).

I thought there would be an option to add automaticaly get Params with every request.

**Reply by yesoreyeram** [reply]
That is coming in next version..

**Reply by mburamorde** [reply]
Is there a chance for a sponsorship for this feature? It would help us enormous ....

**Reply by yesoreyeram** [reply]
v0.8.0-beta.1 is available with this feature. Feel free to test and report issues.

New version also have support for UQL . Give a try there too. ,👍
